src/TurboMqtt/Protocol/Mqtt311Decoder.cs

```diff
@@ -57,9 +57,18 @@ public bool TryDecode(in ReadOnlyMemory<byte> additionalData, out ImmutableList<
             packetType = (MqttPacketType)(currentPacket[0] >> 4);
             currentPacket = currentPacket.Slice(1);
             
-            // extract packet size (the packet span will automatically advance past the size header)
-            if (!TryGetPacketLength(ref currentPacket, out packetSize))
-                return rValue; // we need more data to decode the packet size
+            try
+            {
+                // extract packet size (the packet span will automatically advance past the size header)
+                if (!TryGetPacketLength(ref currentPacket, out packetSize))
+                {
+                    return rValue; // we need more data to decode the packet size
+                }
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Decoded: {packets.Count}, Failing header: {BitConverter.ToString(workingBuffer[..6].ToArray())}", e);
+            }
```

Comment:
Debug code, if the package size is malformed, include the header bytes in the exception message